src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.cs

```diff
@@ -1020,6 +1054,9 @@ public virtual void BeginCatchBlock(Type exceptionType)
             }
 
             current.MarkCatchAddr(m_length, exceptionType);
+
+            // Stack depth for "catch" starts at one.
+            m_curDepth = 1;
```

Comment:
If this doing a proper accounting for exception handlers, can this be simplified?

https://github.com/dotnet/runtime/blob/41419131095d36fb5b811600ad0dab3b0d804269/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/MethodBuilder.cs#L312

Reply:
MethodBuilder and ILGenerator are tightly coupled. I think this should be done in this PR.